src/app/api/index.ts

```diff
@@ -319,44 +322,82 @@ class Client implements API {
         return newAssignment;
     }
 
-    async createAssignmentTemplate(newTemplate: SheriffAssignmentTemplate): Promise<SheriffAssignmentTemplate> {
-        await randomDelay();
-        let assignment = newTemplate.assignmentTemplate;
-
-        //This is a hack to create a unique id for a new assignment template
-        newTemplate.id = defaultAssignmentTemplates.length;
-        assignment.id = defaultAssignmentTemplates.length;        
+    private setAssignmentTemplateTitle(assignment: Partial<SheriffAssignment>): string {
```

Comment:
in this case, we're getting a title for a given `SheriffAssignment`

Perhaps this could be:
```ts
private getAssignmentTitle(assignment: Partial<SheriffAssignment>):string{
...
```

If you need a default parameter, just add it into your function and give it a default value:

```ts
private getAssignmentTitle(assignment: Partial<SheriffAssignment>,defaultTitle:string="Assignment"):string{
...
```

Then later you if you're dealing with an `AssignmentTemplate` you can pass in a sensible default name for it.